File: 0341-flatten-nested-list-iterator/0341-flatten-nested-list-iterator.py
# Define the NestedInteger class if it's not already defined

class NestedIterator:
    def __init__(self, nestedList: List[NestedInteger]):
        self.nodes = list(reversed(nestedList))
    
    def next(self) -> int:
        return self.nodes.pop().getInteger()

    def hasNext(self) -> bool:
        while self.nodes:
            curr = self.nodes[-1]
            if curr.isInteger():
                return True
            self.nodes.pop()
            self.nodes.extend(reversed(curr.getList()))
        return False

File: 0341-flatten-nested-list-iterator/0341-flatten-nested-list-iterator.js
// Define the NestedIterator class if it's not already defined

class NestedIterator {
    constructor(nestedList) {
        this.nodes = [];
        for (let i = nestedList.length - 1; i >= 0; i--) {
            this.nodes.push(nestedList[i]);
        }
    }

    next() {
        return this.nodes.pop().getInteger();
    }

    hasNext() {
        while (this.nodes.length > 0) {
            const current = this.nodes[this.nodes.length - 1];
            if (current.isInteger()) {
                return true;
            }
            this.nodes.pop();
            const adjs = current.getList();
            for (let i = adjs.length - 1; i >= 0; i--) {
                this.nodes.push(adjs[i]);
            }
        }
        return false;
    }
}
